ui_components/src/home/Home.js

```diff
@@ -0,0 +1,564 @@
+import { apiFetch } from "../utils/network.js";
+import type { NewsPostType } from "./newsSchema.js";
+
+
+type Props = {|
+    +newsApiLink: string,
+    +loginErrorText: string,
+    +registerErrorText: string,
+    +resetErrorText: string,
+    +loginMessageText: string,
+    +registerPreFill: $ReadOnlyArray,
+    +initialNewsPosts: $ReadOnlyArray < NewsPostType >,
+|};
+function Home({
+    newsApiLink,
+    loginErrorText,
+    registerErrorText,
+    resetErrorText,
+    loginMessageText,
+    registerPreFill,
+    initialNewsPosts,
+}: Props) {
+    const [displayLogin, setDisplayLogin] = React.useState((loginErrorText != "" || loginMessageText != "") ? true : false);
+    const [displayRegister, setDisplayRegister] = React.useState(registerErrorText == "" ? false : true);
+    const [displayReset, setDisplayReset] = React.useState(resetErrorText == "" ? false : true);
+    const [displayRules, setDisplayRules] = React.useState(false);
+    const [displayTerms, setDisplayTerms] = React.useState(false);
+    const [newsPosts, setNewsPosts] = React.useState(initialNewsPosts);
```

Comment:
None of these are used in this component itself - Could we move them down to the `LoginSection` component and avoid a bunch of extra props?

Reply:
Marking as to-do.